server/src/main/java/org/gluu/oxtrust/api/RegistrationManagementService.java

```diff
@@ -0,0 +1,211 @@
+    private AttributeService attributeService;
+
+    @Inject
+    private JsonConfigurationService jsonConfigurationService;
+
+
+    /**
+     * This operation retrieves the configuration info for the specified findAttributesForPattern criteria,
+     * or all the configurations if no findAttributesForPattern criteria specified.
+     * <p>
+     * Note:
+     * Due to some properties of {@link GluuAttribute} class, direct JSON mapping may result in JSON exceptions.
+     * This may not be an issue on browser side, however, the java clients that are calling this method
+     * should not rely on automatic conversion of JSON objects to {@link RegistrationManagementResponse} class
+     * and should rather use {@link org.codehaus.jackson.map.DeserializationConfig.Feature} class <code>FAIL_ON_UNKNOWN_PROPERTIES</code> properties
+     * for deserialization.
+     *
+     * @param searchPattern   Search pattern.
+     * @param securityContext An injectable interface that provides access to security related information.
+     * @return response
+     */
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @ApiOperation(
+            value = "Reads registration configuration info.",
+            notes = "Reads registration configuration info."
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 403, message = "access_denied\n" +
+                    SERVER_DENIED_THE_REQUEST),
+            @ApiResponse(code = 200, response = RegistrationManagementResponse.class, message = "Success")
+    })
+    public Response getConfiguration(@QueryParam(value = "searchPattern") String searchPattern, @HeaderParam("Authorization") String authorization, @Context HttpServletRequest httpRequest, @Context SecurityContext securityContext) {
+        log.debug("Attempting to read configurations: searchPattern = {}", searchPattern);
+        Response.ResponseBuilder builder = search(searchPattern);
+        Response response = builder.build();
+        return response;
+    }
+
+    /**
+     * Saves the registration configuration data
+     *
+     * @param requestJson     request parameters
+     * @param httpRequest     http request object
+     * @param securityContext An injectable interface that provides access to security related information.
+     * @return response
+     */
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @ApiOperation(
+            value = "Saves configuration.",
+            notes = "Saves configuration.",
+            response = Response.class
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 400, message = "invalid_request\n" +
+                    MALFORMED_REQUEST),
+            @ApiResponse(code = 403, message = "access_denied\n" +
+                    SERVER_DENIED_THE_REQUEST)
+
+    })
+    public Response saveConfiguration(String requestJson, @HeaderParam("Authorization") String authorization, @Context HttpServletRequest httpRequest, @Context SecurityContext securityContext) {
```

Comment:
Same as the method above.
Please don't take the json text as input here, instead use the RegistrationManagementRequest.

Reply:
We have to use the raw json as some of the Fields of attribute class are private and they abrupt JSON DE-serialization.